core/src/components/button/button.ionic.scss

```
@@ -53,45 +53,59 @@
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-solid.ion-color) {
+  --background-activated: #{globals.current-color(shade)};
+  --background-hover: #{globals.current-color(shade)};
+  --background: #{globals.current-color(base)};
+  --color: #{globals.current-color(contrast)};
+}
+
 // Outline Button
 // --------------------------------------------------
 
 :host(.button-outline) {
   --border-width: #{globals.$ion-border-size-025};
   --border-style: #{globals.$ion-border-style-solid};
-  --background-activated: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-activated: #{globals.ion-color(primary, shade, $subtle: true)};
   --background-focused: transparent;
-  --background-hover: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-hover: #{globals.ion-color(primary, shade, $subtle: true)};
   --background-hover-opacity: 1;
   --border-color: #{globals.ion-color(primary, base)};
   --color: #{globals.ion-color(primary, base)};
   --ripple-opacity: var(--background-activated-opacity, 1);
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-outline.ion-color) {
+  --background-activated: #{globals.current-color(shade, $subtle: true)};
+  --background-hover: #{globals.current-color(shade, $subtle: true)};
+  --border-color: #{globals.current-color(base)};
+  --color: #{globals.current-color(base)};
+}
+
+// Warning Outline Button - use foreground color for text and border
+:host(.button-outline.ion-color-warning) .button-native {
+  border-color: #{globals.current-color(foreground)};
+
+  color: #{globals.current-color(foreground)};
```

We should be using the warning color since we're targeting warning. Also we don't need to use `#{}` when using a property.




```suggestion
  border-color: globals.ion-color(warning, foreground);
  color: globals.ion-color(warning, foreground);
```